scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -163,192 +42,120 @@
+          {:expected-state :expected/starting :launch-pod (api/task-metadata->pod task-metadata)}))))
 
   (kill-task [this task-id]
-    (throw (UnsupportedOperationException. "Cannot kill tasks")))
+    (controller/update-expected-state this task-id {:expected-state :expected/killed}))
 
-  (decline-offer [this offer-id])
+  (decline-offers [this offer-ids]
+    (log/debug "Rejecting offer ids" offer-ids))
 
   (db-id [this]
     entity-id)
 
   (compute-cluster-name [this]
     name)
 
-  (initialize-cluster [this pool->fenzo pool->offers-chan]
+  (initialize-cluster [this pool->fenzo running-task-ents]
+    ; Initialize the pod watch path.
     (let [conn cook.datomic/conn
-          handle-exit-code (fn handle-exit-code [task-id exit-code]
-                             (sandbox/aggregate-exit-code exit-code-syncer-state task-id exit-code))
-          handle-status-update (fn handle-status-update [status]
-                                 (scheduler/handle-status-update conn
-                                                                 this
-                                                                 pool->fenzo
-                                                                 (constantly nil) ; no sandboxes to sync
-                                                                 status))
-          pod-callback (make-pod-watch-callback handle-status-update handle-exit-code)]
-      (initialize-pod-watch api-client current-pods-atom pod-callback))
-    (initialize-node-watch api-client current-nodes-atom)
+          pod-callback (make-pod-watch-callback this)]
+      (api/initialize-pod-watch api-client current-pods-atom pod-callback)
+      ; We require that initialize-pod-watch sets current-pods-atom before completing.
+      (reset! expected-state-map (determine-expected-state conn name running-task-ents current-pods-atom)))
+
+    ; Initialize the node watch path.
+    (api/initialize-node-watch api-client current-nodes-atom)
+    ; TODO: Need to visit every state to refresh (i.e., do a single pass of state scanner)
+
+    (reset! pool->fenzo-atom pool->fenzo)
 
-    ; TODO(pschorf): Figure out a better way to plumb these through
-    (chime/chime-at (tp/periodic-seq (t/now) (t/seconds 2))
-                    (fn [_]
-                      (try
-                        (let [nodes @current-nodes-atom
-                              pods @current-pods-atom
-                              offers (generate-offers nodes pods this)
-                              pool (config/default-pool)
-                              chan (pool->offers-chan pool)] ; TODO(pschorf): Support pools
-                          (log/info "Processing offers:" offers)
-                          (scheduler/receive-offers chan match-trigger-chan this pool offers))
-                        (catch Exception e
-                          (log/error e "Exception while forwarding offers")))))
     ; TODO(pschorf): Deliver when leadership lost
```

Comment:
This still a TODO?

Reply:
Can you replace it with a comment saying that we're keeping leadership indefinitely in Kubernetes?

Reply:
I did, but I don't see the connection between that and the async chan. Can you revise this? Maybe something like 'because we keep leadership indefinitely, we never send on this channel to indicate leadership lost'?